.github/workflows/publish_pypi.yml

```diff
@@ -8,13 +8,13 @@ on:
 
 jobs:
   build_and_upload:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
     - uses: actions/checkout@master
-    - name: Set up Python 3.7
+    - name: Set up Python 3.12
       uses: actions/setup-python@master
       with:
-        python-version: '3.7'
+        python-version: '3.12'
```

Comment:
Is there any reason why you didn't use Python 3.13?

Reply:
I taught its in prerelease stage. Since its released and in bugfix stage, I'll update it.